pkg/game/board.go

```diff
@@ -73,9 +72,23 @@ func NewBoard(tileSet tilesets.TileSet) elements.Board {
 func (board board) DeepClone() elements.Board {
 	// note: skipped board.tileSet because TileSet is immutable
 
-	// PlacedTile is not mutated after it's placed
-	board.tiles = slices.Clone(board.tiles)
-	board.tilesMap = maps.Clone(board.tilesMap)
+	tilesMap := map[position.Position]elements.PlacedTile{}
+	tiles := make([]elements.PlacedTile, len(board.tileSet.Tiles)+1)
+
+	for pos, tile := range board.tilesMap {
+		tile = tile.DeepClone()
+		tilesMap[pos] = tile
+	}
+	board.tilesMap = tilesMap
+
+	tiles[0] = board.tiles[0]
+	for i, tile := range board.tiles {
+		if tile.Position.X() == 0 && tile.Position.Y() == 0 {
+			continue
+		}
```

Comment:
Is it possible to simply check if i==0, or before loop create shallow copy, without first tile? To ignore checking if starting tile in every iteration.

Reply:
That's, surprisingly, not the point of this check - the `Tiles` slice is actually of fixed size (was meant to be used as a vector for the agent) and not all tiles in it are real. This is one way of checking it though there are probably some simpler ones - I could just check if `tile.Features == nil`. Either way, I should probably add a comment for this.